core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -124,16 +124,35 @@ public String generateLoadBalancerEndpoint(Stack stack) {
         }
     }
 
+    /*
+     * favors public over private
+     */
     public String getLoadBalancerUserFacingFQDN(Long stackId) {
         Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId).stream()
-            .filter(lb -> StringUtils.isNotEmpty(lb.getFqdn()))
-            .collect(Collectors.toSet());
-        LoadBalancer loadBalancer = loadBalancers.stream()
-            .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
-            .findAny()
-            .orElseGet(() -> loadBalancers.stream().findAny().orElse(null));
+                .filter(lb -> StringUtils.isNotEmpty(lb.getDns()) ||
+                        StringUtils.isNotEmpty(lb.getIp()) || StringUtils.isNotEmpty(lb.getFqdn()))
+                .collect(Collectors.toSet());
+        return loadBalancers.stream()
```

Comment:
it's much better than previous, I would just move out the 2 methods to make things a bit more nice:
```
    public String getLoadBalancerUserFacingFQDN(Long stackId) {
        Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId).stream()
                .filter(lb -> StringUtils.isNotBlank(lb.getDns()) || StringUtils.isNotBlank(lb.getIp()) || StringUtils.isNotBlank(lb.getFqdn()))
                .collect(Collectors.toSet());
        return findPublicLbName(loadBalancers)
                .orElseGet(() -> findPrivateLbNameOrNull(loadBalancers));
    }